designs/odcr.md

```diff
@@ -0,0 +1,408 @@
+
+### Capacity Reservations
+
+Each [Capacity Reservation](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2@v1.162.1/types#CapacityReservation) is defined with:
+
+- The Availability Zone in which to reserve the capacity
+- The count of instances for which to reserve capacity
+- The instance attributes, including the instance type, tenancy, and platform/OS
+- Instance match criteria
+  - Targeted -- only accept instances that matches all attributes + explicitly targeted the capacity reservation
+  - Open -- if capacity reservation accepts all instances that matches all attributes
+- Reservation type
+  - [Default](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-capacity-reservations.html) -- standard capacity reservations, pre-reserved capacity for on-demand instances in arbitrary instance counts
+  - [Capacity Block](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/capacity-blocks-using.html) -- pre-reserved capacity in specific block sizes that is only allocated for a specific amount of time (up to 14 days in 1-day increments or up to 28 days in 7-day increments)
+- A start and end date (if applicable) for when the reservation of capacity is available
+
+AWS also supports grouping Capacity Reservation into [Capacity Reservation groups](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-cr-group.html). Both these entities are supported in Launch Template's CapacityReservationTarget [definitions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-capacityreservationtarget.html).
+
+## Goals
+
+1. Allow selection of targeted and open ODCRs with Karpenter 
+2. Ensure multiple ODCRs can be selected from a single NodePool 
+3. Ensure that we only launch capacity into an ODCR in a cluster when an application requires the capacity, ensuring ODCR sharing between clusters and accounts
+4. Ensure ODCRs are prioritized over regular OD and spot capacity 
+5. Ensure Karpenter consolidates regular OD and spot instances to ODCR capacity when it is available 
+6. Ensure Karpenter consolidates between ODCRs when a smaller/cheaper ODCR is available
+7. Allow users to constrain a NodePool to only launch into ODCR capacity without fallback 
+8. Allow users to fallback from ODCR to spot capacity and from ODCR to standard OD capacity 
+9. Ensure OD capacity is not automatically drifted to new capacity when a capacity reservation expires or is canceled to reduce workload disruption
```

Comment:
It will be consolidated if there are capacity reservations available but the instances shouldn't be drifted just be virtue of it not being a capacity reservation so long as the NodePool supports on-demand instances as well as reserved instances.

Reply:
Ah just the last part was missing for me. If there's no fallback, it should still be drifted.